doc/source/user_guide/text.rst

```diff
@@ -29,7 +34,16 @@ for many reasons:
 3. When reading code, the contents of an ``object`` dtype array is less clear
    than ``'string'``.
 
-Currently, the performance of ``object`` dtype arrays of strings and
+Starting from pandas 3.0, the default dtype for string data is ``str``. Previously, ``object`` dtype was the
+default for string data. You may still encounter ``object`` dtype for legacy datasets or while working with earlier
+pandas versions. Use ``.astype("str")`` to explicitly convert these to ``str`` dtype or specify ``dtype="str"``
+when creating new data structures.
+
+Use the nullable :class:`StringDtype` (``"string"``) or ``str`` dtype when handling NA values in your string data.
+Note that ``StringDtype`` uses ``pd.NA`` for missing values, whereas ``str`` dtype uses ``np.NaN``. ``StringDtype``
```

Comment:
This is incorrect, since `StringDtype` accepts an `na_value` argument to choose between `pd.NA` and `np.nan`. `np.NaN` has been removed as of NumPy 2.0, so shouldn't be used.